File: modules/agent/src/models/Deal.py
"""Deal model."""

from sqlalchemy import Column, Text, DateTime, BigInteger, Numeric, Date, ForeignKey, func, Index
from sqlalchemy.orm import relationship

from models import Base


class Deal(Base):
    """Deal SQLAlchemy model."""

    __tablename__ = "Deal"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    tenant_id = Column(BigInteger, ForeignKey("Tenant.id", ondelete="CASCADE"), nullable=True)
    name = Column(Text, nullable=False)
    description = Column(Text, nullable=True)
    owner_individual_id = Column(BigInteger, ForeignKey("Individual.id", ondelete="SET NULL"), nullable=True)
    current_status_id = Column(BigInteger, ForeignKey("Status.id", ondelete="SET NULL"), nullable=True)
    value_amount = Column(Numeric(18, 2), nullable=True)
    value_currency = Column(Text, default="USD")
    probability = Column(Numeric(5, 2), nullable=True)
    expected_close_date = Column(Date, nullable=True)
    close_date = Column(Date, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    tenant = relationship("Tenant", back_populates="deals")
    current_status = relationship("Status", back_populates="deals")
    leads = relationship("Lead", back_populates="deal")

    __table_args__ = (
        Index('idx_deal_tenant_id', 'tenant_id'),
    )

File: modules/agent-go/internal/models/Deal.go
package models

import (
	"time"

	"github.com/shopspring/decimal"
)

type Deal struct {
	ID                int64            `gorm:"primaryKey;autoIncrement" json:"id"`
	TenantID          *int64           `gorm:"index" json:"tenant_id"`
	ProjectID         *int64           `gorm:"index" json:"project_id"`
	Name              string           `gorm:"not null" json:"name"`
	Description       *string          `json:"description"`
	OwnerIndividualID *int64           `json:"owner_individual_id"`
	CurrentStatusID   *int64           `json:"current_status_id"`
	ValueAmount       *decimal.Decimal `gorm:"type:numeric(18,2)" json:"value_amount"`
	ValueCurrency     string           `gorm:"default:USD" json:"value_currency"`
	Probability       *int16           `json:"probability"` // 0-100
	ExpectedCloseDate *time.Time       `json:"expected_close_date"`
	CloseDate         *time.Time       `json:"close_date"`
	CreatedAt         time.Time        `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt         time.Time        `gorm:"autoUpdateTime" json:"updated_at"`
	CreatedBy         int64            `gorm:"not null" json:"created_by"`
	UpdatedBy         int64            `gorm:"not null" json:"updated_by"`
}

func (Deal) TableName() string {
	return "Deal"
}
